game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtySortingUtil.java

```diff
@@ -10,46 +10,24 @@
 
 @UtilityClass
 public class CasualtySortingUtil {
-  /**
-   * In an amphibious assault, sort on who is unloading from transports first as this will allow the
-   * marines with higher scores to get killed last.
-   */
-  public static void sortAmphib(final List<Unit> units, final List<Unit> amphibiousLandAttackers) {
-    final Comparator<Unit> decreasingMovement =
-        UnitComparator.getLowestToHighestMovementComparator();
+  /** In an amphibious assault, sort marines with higher scores last */
+  public static void sortPreBattle(final List<Unit> units) {
     units.sort(
         Comparator.comparing(Unit::getType, Comparator.comparing(UnitType::getName))
-            .thenComparing(
-                (u1, u2) -> {
-                  final UnitAttachment ua = UnitAttachment.get(u1.getType());
-                  final UnitAttachment ua2 = UnitAttachment.get(u2.getType());
-                  if (ua.getIsMarine() != 0 && ua2.getIsMarine() != 0) {
-                    return compareAccordingToAmphibious(u1, u2, amphibiousLandAttackers);
-                  }
-                  return 0;
-                })
-            .thenComparing(decreasingMovement));
+            .thenComparing(compareMarines())
+            .thenComparing(UnitComparator.getLowestToHighestMovementComparator()));
   }
 
-  private static int compareAccordingToAmphibious(
-      final Unit u1, final Unit u2, final List<Unit> amphibiousLandAttackers) {
-    if (amphibiousLandAttackers.contains(u1) && !amphibiousLandAttackers.contains(u2)) {
-      return -1;
-    } else if (amphibiousLandAttackers.contains(u2) && !amphibiousLandAttackers.contains(u1)) {
-      return 1;
-    }
-    final int m1 = UnitAttachment.get(u1.getType()).getIsMarine();
-    final int m2 = UnitAttachment.get(u2.getType()).getIsMarine();
-    return m2 - m1;
-  }
+  public static Comparator<Unit> compareMarines() {
```

Comment:
Before I changed this class, I wrote some unit tests to make sure I understood the logic.  And I got unexpected results.  So, I "fixed" the class to match the tests.  Take a look at the tests to see if maybe I have an incorrect assumption on how this is supposed to sort units.

Also, as I mentioned earlier, I'm not sure this sort code is even needed.